echopype/mask/api.py

```diff
@@ -0,0 +1,306 @@
+        The pair of nominal frequencies to be used for frequency-differencing, where
+        the first element corresponds to ``freqA`` and the second element corresponds
+        to ``freqB``
+    chanAB: list of float, optional
+        The pair of channels that will be used to select the nominal frequencies to be
+        used for frequency-differencing, where the first element corresponds to ``freqA``
+        and the second element corresponds to ``freqB``
+
+    Returns
+    -------
+    source_Sv: xr.Dataset
+        A Dataset containing the Sv data
+    """
+
+    source_Sv, file_type = validate_source_Sv(source_Sv, storage_options)
+
+    if isinstance(source_Sv, str):
+
+        # open up Dataset using source_Sv path
+        source_Sv = xr.open_dataset(source_Sv, engine=file_type, chunks="auto", **storage_options)
+
+    # check that channel and frequency nominal are in source_Sv
+    if "channel" not in source_Sv.coords:
+        raise RuntimeError("The Dataset defined by source_Sv must have channel as a coordinate!")
+    elif "frequency_nominal" not in source_Sv.variables:
+        raise RuntimeError(
+            "The Dataset defined by source_Sv must have frequency_nominal as a variable!"
+        )
+
+    # make sure that the channel and frequency_nominal values are not repeated in source_Sv
+    if len(set(source_Sv.channel.values)) != 2:
+        raise RuntimeError(
+            "The provided source_Sv contains repeated channel values, " "this is not allowed!"
+        )
+
+    if len(set(source_Sv.frequency_nominal.values)) != 2:
+        raise RuntimeError(
+            "The provided source_Sv contains repeated frequency_nominal "
+            "values, this is not allowed!"
+        )
+
+    def get_positions(var_name, input_list):
+
+        # obtain position of input in either frequency_nominal or channel
+        inputA_pos = np.argwhere(source_Sv[var_name].values == input_list[0]).flatten()
+        inputB_pos = np.argwhere(source_Sv[var_name].values == input_list[1]).flatten()
+
+        return inputA_pos, inputB_pos
+
+    # get the position of freqAB or chanAB values in frequency_nominal or channel, respectively
+    if freqAB is not None:
+        inputA_pos, inputB_pos = get_positions(var_name="frequency_nominal", input_list=freqAB)
+    else:
+        inputA_pos, inputB_pos = get_positions(var_name="channel", input_list=chanAB)
+
+    # check that the elements of freqAB or chanAB are in frequency_nominal or channel, respectively
+    if (len(inputA_pos) == 0) or (len(inputB_pos) == 0):
+
+        if freqAB is not None:
+            raise RuntimeError(
+                "The provided list input freqAB contains values that "
+                "are not in the frequency_nominal variable!"
+            )
+        else:
+            raise RuntimeError(
+                "The provided list input chanAB contains values that are "
+                "not in the channel coordinate!"
+            )
```

Comment:
Why do you need to determine the position to know whether the elements of `freqAB` or `chanAB` are in the dataset? Wouldn't that just take a list comprehension like
```
np.all([f in souce_Sv["frequency_nominal] for f in freqAB])
```
and equivalent for `channel` and raise RuntimeError when the above is `False`?

Reply:
You are right, there is a better way to implement this using list comprehension. I previously was utilizing `get_positions` such that this made sense, but I ended up removing it. I will go ahead and improve this using your suggestions.